extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java

```diff
@@ -278,6 +227,106 @@ private void consumeService(Class<?> serviceClass, BiConsumer<Class<?>, Collecti
         }
     }
 
+    private void resolveLiquibaseResources(
+            BuildProducer<ServiceProviderBuildItem> services,
+            BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized,
+            BuildProducer<NativeImageResourceBuildItem> resource) {
+
+        var jarUrlTemplate = "jar:%s!/";
+        try {
+            var srcUrl = Objects.requireNonNull(Liquibase.class
+                    .getProtectionDomain()
+                    .getCodeSource()
+                    .getLocation());
+            var jarUrl = new java.net.URL(jarUrlTemplate.formatted(srcUrl.toString())).toURI();
+
+            try (var ignored = FileSystems.newFileSystem(jarUrl, Collections.EMPTY_MAP)) {
+                var rootPath = Paths.get(jarUrl);
+                loadLiquibaseServiceProviderConfig(rootPath, services, reflective, runtimeInitialized);
+                loadLiquibaseRootProperties(rootPath, resource);
+                loadLiquibaseXsdResources(rootPath, resource);
+            }
+        } catch (URISyntaxException | IOException ex) {
+            throw new IllegalStateException(ex);
+        }
```

Comment:
I'm not entirely sure about the implementation here.

I think, maybe using `CurateOutcomeBuildItem` to get the `ApplicationModel` would be better. Then you could get a `ResolvedDependency` from `getDependencies` that match Liquibase and use `getContentTree(PathFilter)`.

Reply:
Thanks for the pointer. Done.